include/envoy/network/connection.h

```diff
@@ -69,6 +69,37 @@ enum class ConnectionCloseType {
                      // until the delayed_close_timeout expires
 };
 
+/**
+ * Run-time modifiable connection options to alter socket settings
+ */
+class DynamicSocketOptions {
+public:
+  virtual ~DynamicSocketOptions() = default;
+
+  /**
+   * Set the receive buffer size for the socket to a value specified by the 'buffer_size'
+   */
+  virtual void setSocketRecvBufferSize(uint32_t buffer_size) const PURE;
```

Comment:
I'd assume these should all return SyscallIntResults, so one can check on success / failure

Reply:
Got it. I will make this change.